src/ucs/sys/sock.c

```diff
@@ -171,14 +173,30 @@ unsigned ucs_netif_bond_ad_num_ports(const char *bond_name)
 {
     ucs_status_t status;
     long ad_num_ports;
-
+    const char *lower_bond_path;
+    char lower_bond_path_buf[PATH_MAX];
+    
     status = ucs_read_file_number(&ad_num_ports, 1,
                                   UCS_NETIF_BOND_AD_NUM_PORTS_FMT, bond_name);
     if ((status != UCS_OK) || (ad_num_ports <= 0) ||
         (ad_num_ports > UINT_MAX)) {
-        ucs_trace("failed to read from " UCS_NETIF_BOND_AD_NUM_PORTS_FMT ": %m, "
-                 "assuming 802.3ad bonding is disabled", bond_name);
-        return 1;
+        lower_bond_path = ucs_topo_iface_get_bond_lower_dev_path(bond_name, UCS_NETIF_CLASS_DIR,
+                                                                 lower_bond_path_buf, PATH_MAX);
+        if (lower_bond_path) {
+            status = ucs_read_file_number(&ad_num_ports, 1, "%s/%s", lower_bond_path,
+                                          UCS_BONDING_AD_NUM_PORTS);
+
+            if ((status != UCS_OK) || (ad_num_ports <= 0) ||
+                (ad_num_ports > UINT_MAX)) {
+                ucs_trace("failed to read from " UCS_NETIF_BOND_AD_NUM_PORTS_FMT " and %s: %m, "
+                          "assuming 802.3ad bonding is disabled", bond_name, lower_bond_path);
+                return 1;
+            }
+        } else {
+            ucs_trace("failed to read from " UCS_NETIF_BOND_AD_NUM_PORTS_FMT ": %m, "
+                      "assuming 802.3ad bonding is disabled", bond_name);
+            return 1;
+        }
```

Comment:
now you can call `ucs_topo_iface_get_sysfs_path` instead of `ucs_topo_iface_get_bond_lower_dev_path`

Reply:
Hi @shasson5, looks ucs_topo_iface_get_sysfs_path doesn't fit for my fix.  My fix it to find the lower bond in vlan(not PF path). So it doesn't need `sysfs_path = ucs_topo_resolve_sysfs_path(ucs_string_buffer_cstr(&dev_path), path_buffer);`. 

My code logic is to find `lower_bond*` firstly, if not found, then find `lower_*`.

Reply:
I also use loop to deep search `bonding/ad_num_ports`. Thanks.

Reply:
can you refactor the deep search part out of `ucs_topo_iface_get_sysfs_path`?
then you will be able to do the following in `ucs_netif_bond_ad_num_ports`:
```
dev_name = deep_search(bond_name);
lower_bond_path = ucs_dirname(dev_name);
```

Reply:
@shasson5 Thanks for your comments. I move out the deep search code to a separate function. Looks the code is clearer. Please review the new commits. Thanks.